### DISCUSSION TITLE
Code editors and white space

### DISCUSSION THREAD

**Discussion by togfoxy** [togfoxy/MarsLander#177] [General]
I'm struggling to work out the expectation regarding white space between functions and at end of line.

I get the feeling our editors are doing different things so when I can see what should be happening I can see what my editor can and can't do.

**Comment by MadByteDE** [comment]
https://user-images.githubusercontent.com/4752038/140633622-9f2da44a-e4bf-40a6-9904-2de788fd1636.mp4

VSCode and other IDEs do show where whitespaces are located. Naturally when writing code the editor makes sure to not leave whitespace when writing code and if you add some it shows you this markes. All there is to do is remove the markers for us.

Did you try the option I linked to in the issue yesterday? If it doesn't work you might need to just make sure yourself but since they are whitespace they will be hard to spot.

I have no problem with removing your whitespace over time when cleaning up stuff. For PRs it means to have an extra commit afterwards to do this.

edit. damn no mouse in the video, but you get the idea.. the two dots always are whitespace.

**Reply by Philbywhizz** [reply]
I use the vscode extention called [Trailing Spaces](https://marketplace.visualstudio.com/items?itemName=shardulm94.trailing-spaces) as it puts a nice red block when I find trailing spaces

![example](https://shardulm94.gallerycdn.vsassets.io/extensions/shardulm94/trailing-spaces/0.3.1/1554790489790/Microsoft.VisualStudio.Services.Icons.Default)

**Comment by togfoxy** [comment]
There is a 'manual' method in notepad++ that I have to do
EVERY
SINGLE
TIME.

:(

When I work out exactly what it is my notepad needs to do then I'll see if I can find a setting or automate it.

I'll leave this discussion here for now.

**Reply by Philbywhizz** [reply]
Might be a contentious issue, but have you looked trying another editor that might help? (I use Visual Studio Code on both my linux box and my mac)
